File: app/src/main/java/com/dsy/mvp/ui/fragment/SimpleFragment.java
package com.dsy.mvp.ui.fragment;

import android.os.Bundle;

import com.dsy.mvp.R;
import com.dsy.mvp.base.BaseFragment;
import com.dsy.mvp.base.impl.IPresenter;

public class SimpleFragment extends BaseFragment {
    private static final String ARG_PARAM1 = "param1";
    private static final String ARG_PARAM2 = "param2";

    private String mParam1;
    private String mParam2;

    public static SimpleFragment newInstance(String param1, String param2) {
        SimpleFragment fragment = new SimpleFragment();
        Bundle args = new Bundle();
        args.putString(ARG_PARAM1, param1);
        args.putString(ARG_PARAM2, param2);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mParam1 = getArguments().getString(ARG_PARAM1);
            mParam2 = getArguments().getString(ARG_PARAM2);
        }
    }


    @Override
    public int createLayoutId() {
        return R.layout.fragment_simple;
    }

    @Override
    protected IPresenter createPresenter() {
        return null;
    }

    @Override
    protected void initView() {

    }

    @Override
    protected void initEvent() {

    }

    @Override
    protected void initData() {

    }
}

File: app/src/main/java/com/dsy/mvp/ui/fragment/SimpleFragment.kt
package com.dsy.mvp.ui.fragment

import android.os.Bundle
import com.dsy.mvp.R
import com.dsy.mvp.base.BaseFragment
import com.dsy.mvp.base.SimplePresenter

class SimpleFragment : BaseFragment<SimplePresenter>(R.layout.fragment_simple) {

    override fun initView() {}
    override fun initEvent() {}
    override fun initData() {}

    companion object {
        fun newInstance(): SimpleFragment {
            val fragment = SimpleFragment()
            val args = Bundle()
            fragment.arguments = args
            return fragment
        }
    }
}